logstash-core/lib/logstash/pipelines_registry.rb

```diff
@@ -72,14 +79,20 @@ def create_pipeline(pipeline_id, pipeline, &create_block)
     #
     # @yieldparam [Pipeline] the pipeline to terminate
     def terminate_pipeline(pipeline_id, &stop_block)
-      @states.compute(pipeline_id) do |_, state|
```

Comment:
A minor syntatic sugar for this sort of pattern, is that we can have the ensure block at method definition level since lock won't be nil, overall reducing an indentation level. For example in the `terminate_pipeline`:

```ruby
    def terminate_pipeline(pipeline_id, &stop_block)
      lock = get_lock(pipeline_id)
      lock.lock

      state = @states.get(pipeline_id)
      if state.nil?
        logger.error("Attempted to terminate a pipeline that does not exists", :pipeline_id => pipeline_id)
        @states.remove(pipeline_id)
      else
        yield(state.pipeline)
        @states.put(pipeline_id, state)
      end
    ensure
      lock.unlock
      # if get_lock could return nil then:
      # lock.unlock if lock
    end
```